pcsx2/DEV9/ATA/ATA_Info.cpp

```diff
@@ -150,7 +150,7 @@ void ATA::CreateHDDinfo(u64 sizeSectors)
 	 */
 	WriteUInt16(identifyData, &index, static_cast<u16>(curMultipleSectorsSetting | (1 << 8))); //word 59
 	//Total number of user addressable logical sectors
-	WriteUInt32(identifyData, &index, static_cast<u32>(nbSectors < 268435456 ? nbSectors : 268435456)); //word 60-61
+	WriteUInt32(identifyData, &index, static_cast<u32>(nbSectors < maxSize ? nbSectors : maxSize)); //word 60-61
```

Comment:
```suggestion
	WriteUInt32(identifyData, &index, static_cast<u32>(nbSectors < ((1 << 28) - 1) ? nbSectors : ((1 << 28) - 1))); //word 60-61
```

In case we add support for LBA48 drives with a higher `maxSize`

Reply:
![image](https://github.com/PCSX2/pcsx2/assets/621640/d43f9dd0-c80d-4a66-a0c5-2dc079afce39)

Reply:
casting to u32 will be enough in all possible scenarios

Reply:
`0FFFFFFF` and `(1 << 28) - 1` equate to the same value (268435455)
If we add support for LBA48 along with larger drives, than we will need to clamp this field to the above limit